packages/legacy-client/src/client/objectSets/createObjectSetSearchAround.ts

```diff
@@ -15,12 +15,51 @@
  */
 
 import type { ObjectTypesFrom, OntologyDefinition } from "@osdk/api";
+import type {
+  ObjectSetDefinition,
+  SearchAroundObjectSetDefinition,
+} from "../../ontology-runtime";
+import type { ClientContext } from "../../ontology-runtime/ontologyProvider/calls/ClientContext";
+import type { ObjectSet } from "../interfaces/objectSet";
 import type { SearchAround } from "../interfaces/searchAround";
 import type { OsdkLegacyObjectFrom } from "../OsdkObject";
+import { createOsdkObjectSet } from "./OsdkObjectSet";
 
 export function createObjectSetSearchAround<
   O extends OntologyDefinition<any>,
   K extends ObjectTypesFrom<O>,
->(): SearchAround<OsdkLegacyObjectFrom<O, K>> {
-  throw new Error("not implemented");
+>(
+  clientContext: ClientContext,
+  sourceApiName: K,
+  objectSet: ObjectSetDefinition,
+  ontologyDefinition: O,
+): SearchAround<OsdkLegacyObjectFrom<O, K>> {
+  const result = {} as SearchAround<OsdkLegacyObjectFrom<O, K>>;
+
+  const objectDefinition = clientContext.ontology.objects[sourceApiName];
+  for (const [link, { targetType }] of Object.entries(objectDefinition.links)) {
+    const key = `searchAround${capitalize(link)}` as keyof typeof result;
+    result[key] = () => {
+      const definition = {
+        type: "searchAround",
+        objectSet,
+        link,
+      } satisfies SearchAroundObjectSetDefinition;
+
+      const objSet = createOsdkObjectSet(
+        clientContext,
+        targetType,
+        definition,
+        ontologyDefinition,
+      );
+
+      return objSet as ObjectSet<O, any>;
```

Comment:
For some reason this isn't satisfying the type on `result[key]` and I haven't figured it out yet